src/workflows/airqo_etl_utils/bigquery_api.py

```
@@ -1301,16 +1307,102 @@ def fetch_device_data_for_forecast_job(
+        """
+        Fetch daily aggregated PM2.5 stats per site for forecast jobs.
+
+        Notes:
+        - Filters to rows with non-null pm2_5_calibrated_value
+        - Requires at least `min_hours` hourly points per day (default 18)
+        - Aggregates mean, min, max PM2.5 and hourly count per site per day
+    """
+        try:
+            min_hours = int(min_hours)
+        except (TypeError, ValueError) as e:
+            raise ValueError(f"min_hours must be an integer, got {min_hours}") from e
```

Ruff (0.15.2)

[warning] Avoid specifying long messages outside the exception class

(TRY003)